sys-process/fcron/fcron-3.3.2.ebuild

```diff
@@ -0,0 +1,254 @@
+	sed -e '/FIFODIR/d' -i Makefile.in || die
+
+	# Workaround for:
+	#  * QA Notice: system executables owned by nonzero uid:
+	# But fcron by design doesn't suid root. Hence, hide that fact
+	# from the QA check via a wrapper (not a symlink)
+	# https://bugs.gentoo.org/925512
+	for file in fcrontab fcrondyn; do
+		cat > "${file}_wrapper" <<-EOF
+		#!/bin/sh
+		exec "${EPREFIX}/usr/libexec/${file}" "\$@"
+		EOF
+	done
+
+	eautoconf
+}
+
+src_configure() {
+	# Don't try to pass --with-debug as it'll play with cflags as
+	# well, and run foreground which is a _very_ nasty idea for
+	# Gentoo.
+	use debug && append-cppflags -DDEBUG
+
+	# bindir is used just for calling fcronsighup
+	local myeconfargs=(
+		--with-cflags="${CFLAGS}"
+		--bindir=/usr/libexec
+		--sbindir=/usr/libexec
+		$(use_with audit)
+		$(use_with mta sendmail)
+		$(use_with pam)
+		$(use_with readline)
+		$(use_with selinux)
+		--sysconfdir=/etc/fcron
+		--with-username=fcron
+		--with-groupname=fcron
+		--with-piddir=/run
+		--with-spooldir=/var/spool/fcron
+		--with-fifodir=/run
+		--with-fcrondyn=yes
+		--disable-checks
+		--with-editor=/usr/libexec/editor
+		--with-shell=/bin/sh
+		--without-db2man
+		--without-dsssl-dir
+		--with-rootname=${rootuser}
+		--with-rootgroup=${rootgroup}
+		--with-systemdsystemunitdir="$(systemd_get_systemunitdir)"
+	)
+
+	econf "${myeconfargs[@]}"
+}
+
+src_compile() {
+	default
+
+	# bug #216460
+	sed \
+		-e 's:/usr/local/etc/fcron:/etc/fcron/fcron:g' \
+		-e 's:/usr/local/etc:/etc:g' \
+		-e 's:/usr/local/:/usr/:g' \
+		-i doc/*/*/*.{txt,1,5,8,html} \
+		|| die "unable to fix documentation references"
+}
+
+src_install() {
+	emake install BOOTINSTALL=0 DESTDIR="${ED}" STRIP=echo
+
+	keepdir /var/spool/fcron
+	fowners fcron:fcron /var/spool/fcron
+	fperms 6770 /var/spool/fcron
+
+	newbin fcrontab_wrapper fcrontab
+	newbin fcrondyn_wrapper fcrondyn
+
+	# bitstring.h is a private header inside fcron, not even installed.
+	find "${ED}/usr/share/man" -name '*bitstring*' -delete || die
+
+	if use system-crontab; then
+		dosym fcrontab /usr/bin/crontab
+
+		exeinto /usr/libexec
+		newexe script/check_system_crontabs.sh check_system_crontabs
+
+		insinto /etc/fcron
+		newins "${FILESDIR}"/fcrontab.2 fcrontab
+
+		fowners ${rootuser}:fcron /etc/fcron/fcrontab
+		fperms 0640 /etc/fcron/fcrontab
+
+		insinto /etc
+		newins "${FILESDIR}"/crontab.2 crontab
+	fi
+
+	rm "${ED}/etc/fcron/pam.conf" || die
+	use pam && pamd_mimic system-services fcron auth account session
+	cat > "${T}"/fcrontab.pam <<- EOF
+	# Don't ask for the user's password; fcrontab will only allow to
+	# change user if running as root.
+	auth		sufficient		pam_permit.so
+
+	# Still use the system-auth stack for account and session as the
+	# sysadmin might have set up stuff properly, and also avoids
+	# sidestepping limits (since fcrontab will run \$EDITOR).
+	account		include			system-auth
+	session		include			system-auth
+	EOF
+	use pam && newpamd "${T}"/fcrontab.pam fcrontab
```

Comment:
These might be better grouped inside `if use pam ; then`... block? Including the `rm` command.